lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -185,13 +195,16 @@ void ExecutionEngine::insertCompiledFunction(
   functionMap[name] = func.get();
   compiledFunctions_[name] = std::move(func);
 
-  std::promise<llvm::Error> addPromise;
-  auto fut = addPromise.get_future();
+  llvm::Error err = MAKE_ERR("Function did not run");
```

Comment:
This will create an unchecked Error because it will be ovewritten in the lambda below. Use a boolean instead